addons/account/models/chart_template.py

```diff
@@ -362,7 +362,9 @@ def _pre_load_data(self, template_code, company, template_data, data):
         e.g. the account codes' width must be standardized to the code_digits applied.
         The fiscal country code must be put in place before taxes are generated.
         """
-        if 'account_fiscal_country_id' in data['res.company'][company.id]:
+        if company.parent_id:
+            fiscal_country = company.parent_id.account_fiscal_country_id
+        elif 'account_fiscal_country_id' in data['res.company'][company.id]:
```

Comment:
I think this isn't the right fix.
Here, we are forcing the branch to have the same fiscal country while we just want it to have the same currency, which is set below on line 382.

I think it's there that we should use the same condition and assign the parent company currency.